fastapi_azure_auth/user.py

```diff
@@ -1,16 +1,264 @@
-from typing import Any, Dict, List, Optional
-
-from pydantic import BaseModel, Field
-
-
-class User(BaseModel):
-    aud: str = Field(..., description='Audience')
-    tid: Optional[str] = Field(default=None, description='Tenant ID')
-    roles: List[str] = Field(default=[], description='Roles (Groups) the user has for this app')
-    claims: Dict[Any, Any] = Field(..., description='The entire decoded token')
-    scp: Optional[str] = Field(default=None, description='Scope')
-    name: Optional[str] = Field(default=None, description='Name')
-    access_token: str = Field(..., description='The access_token. Can be used for fetching the Graph API')
-    is_guest: bool = Field(False, description='The user is a guest user in the tenant')
-    sub: str = Field(..., description='Principal associated with the token.')
-    oid: str = Field(..., description='Immutable identifier for the requestor')
+from typing import Any, Dict, List, Literal, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+
+
+class Claims(BaseModel):
+    """A more complete overview of the claims available in an access token can be found here:
+    https://learn.microsoft.com/en-us/azure/active-directory/develop/access-tokens#payload-claims
+    """
+
+    aud: str = Field(
+        ...,
+        description='Identifies the intended audience of the token. In v2.0 tokens, this value is always the client ID'
+        ' of the API. In v1.0 tokens, it can be the client ID or the resource URI used in the request.',
+    )
+    iss: str = Field(
+        ...,
+        description='Identifies the STS that constructs and returns the token, and the Azure AD tenant of the'
+        ' authenticated user. If the token issued is a v2.0 token (see the ver claim), the URI ends in /v2.0.',
+    )
+    idp: Optional[str] = Field(
+        default=None,
+        description='Records the identity provider that authenticated the subject of the token. This value is identical'
+        " to the value of the Issuer claim unless the user account isn't in the same tenant as the issuer, such as"
+        " guests. Use the value of iss if the claim isn't present.",
```

Comment:
Single and double quote mismatch here

Reply:
pre-commit changes the first one back to single quote

Reply:
Hmm, they're all supposed to be single quotes 😅

Reply:
Ohh, I didn't even see that. Okay I'm good with how it is - sorry!